### PR TITLE
Fix/get interior data cache corner cases

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Rework of GetnteriorData logic. Extract common functions. Fix defects found by unit tests.
Rework unit tests : 
  - Add missed expectations
  - Make ON_CALLs arangments

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc [hunk 1]
```diff
@@ -130,113 +145,306 @@ class GetInteriorVehicleDataRequestTest
...
   NsSmartDeviceLink::NsSmartObjects::SmartObject& msg_params =
       (*mobile_message)[application_manager::strings::msg_params];
+  msg_params[message_params::kModuleType] = mobile_apis::ModuleType::RADIO;
+  application_manager::SharedPtr<
+      rc_rpc_plugin::commands::GetInteriorVehicleDataRequest> command =
+      CreateRCCommand<rc_rpc_plugin::commands::GetInteriorVehicleDataRequest>(
+          mobile_message);
+
+  ON_CALL(mock_policy_handler_, CheckModule(_, _)).WillByDefault(Return(true));
+  // Used empty rc capabilities
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: @LuxoftAKutsan Did not get the idea of this comment... It is related to..???

Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: @AByzhynar it is not clear that rc_capabilities will be used from ON_CALLS frin constructir and it is empty by default. SO I added it as a comment

Comment (AByzhynar) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: @LuxoftAKutsan Confusing comment or place for it... May be add it as condition to test case name?

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc [hunk 2]
```diff
@@ -130,23 +134,143 @@ class GetInteriorVehicleDataRequestTest
...
+
+  command->Run();
+}
+
+TEST_F(
+    GetInteriorVehicleDataRequestTest,
+    Execute_ExpectCorrectMessageSentToHMI_AppSubscribedUnsubscibe_ClearCache) {
+  // Arrange
+  auto module_type = mobile_apis::ModuleType::RADIO;
+  MessageSharedPtr mobile_message = CreateBasicMessage();
```
Comment (ZhdanovP) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: Can we make `module_type` a const?
